Subject: Give a proper Makefile for the upstream package
Author: Michael van der Kolff <mvanderkolff@gmail.com>
Forwarded: no

Upstream Makefile is barebones, and doesn't make any shared libraries.

With some extra effort, this does.
--- a/Makefile
+++ b/Makefile
@@ -11,9 +11,9 @@
 CFLAGS = $(CCFLAGS) -I../libjbig
 
 VERSION=2.0
+.PHONY: all lib pbm test clean install
 
 all: lib pbm
-	@echo "Enter 'make test' in order to start some automatic tests."
 
 lib:
 	(cd libjbig;  make "CC=$(CC)" "CFLAGS=$(CFLAGS)")
@@ -22,17 +22,23 @@
 	(cd pbmtools; make "CC=$(CC)" "CFLAGS=$(CFLAGS)")
 
 test: lib pbm
-	(cd libjbig;  make "CC=$(CC)" "CFLAGS=$(CFLAGS)" test)
-	(cd pbmtools; make "CC=$(CC)" "CFLAGS=$(CFLAGS)" test)
+#	(cd libjbig;  make "CC=$(CC)" "CFLAGS=$(CFLAGS)" test)
+#	(cd pbmtools; make "CC=$(CC)" "CFLAGS=$(CFLAGS)" test)
 
 clean:
 	rm -f *~ core
 	(cd libjbig; make clean)
 	(cd pbmtools; make clean)
 
-distribution: clean
-	rm -f libjbig/libjbig*.a
-	(cd ..; tar -c -v --exclude .svn -f jbigkit-$(VERSION).tar jbigkit ; \
-	  gzip -9f jbigkit-$(VERSION).tar )
-	scp ../jbigkit-$(VERSION).tar.gz slogin-serv1.cl.cam.ac.uk:public_html/download/
-	scp CHANGES slogin-serv1.cl.cam.ac.uk:public_html/jbigkit/
+install: all
+	install -d $(DESTDIR)/usr/lib
+	install -s -m 644 libjbig/.libs/*.so.*.*.* libjbig/.libs/*.a $(DESTDIR)/usr/lib
+	install -m 644 libjbig/.libs/*.la $(DESTDIR)/usr/lib
+	/sbin/ldconfig -n $(DESTDIR)/usr/lib
+	ln -s $(DESTDIR)/usr/lib/libjbig.so.*.* $(DESTDIR)/usr/lib/libjbig.so
+	install -d $(DESTDIR)/usr/include
+	install -m 644 libjbig/*.h $(DESTDIR)/usr/include
+	install -d $(DESTDIR)/usr/bin
+	install -s -m 755 pbmtools/jbgtopbm pbmtools/jbgtopbm85 pbmtools/pbmtojbg pbmtools/pbmtojbg85 $(DESTDIR)/usr/bin
+	install -d $(DESTDIR)/usr/share/man/man1
+	install -m 644 pbmtools/*.1 $(DESTDIR)/usr/share/man/man1
+	install -d $(DESTDIR)/usr/share/man/man5
+	install -m 644 pbmtools/*.5 $(DESTDIR)/usr/share/man/man5
